pkg/scheduler/scheduler.go

```diff
@@ -367,11 +381,48 @@ func (s *Scheduler) forwardRequestToQuerier(querier schedulerpb.SchedulerForQuer
 	case err := <-errCh:
 		// Is there was an error handling this request due to network IO,
 		// then error out this upstream request _and_ stream.
-		// TODO: if err is not nil, scheduler should notify frontend using the frontend address.
+		s.forwardErrorToFrontend(req.ctx, req, err)
 		return err
 	}
 }
 
+func (s *Scheduler) forwardErrorToFrontend(ctx context.Context, req *schedulerRequest, requestErr error) {
+	opts, err := s.cfg.GRPCClientConfig.DialOption([]grpc.UnaryClientInterceptor{
+		otgrpc.OpenTracingClientInterceptor(opentracing.GlobalTracer()),
+		middleware.ClientUserHeaderInterceptor},
+		nil)
+	if err != nil {
+		level.Warn(s.log).Log("msg", "failed to create gRPC connection to frontend to report error", "frontend", req.frontendAddress, "err", err, "requestErr", requestKey{})
+		return
+	}
+
+	conn, err := grpc.DialContext(ctx, req.frontendAddress, opts...)
```

Comment:
[just wondering] Should we just use the usual client pool?

Reply:
I opted not to do that, as I don't expect this to happen too frequently. But I don't have strong opinion on it, and can change to pool if you think it's better.

Reply:
I was just wondering what's your take. I'm fine creating a new connection each time.